### PR TITLE
Prepare release 1.7.0 - Add token to Blackbox Exporter ConfigMap

### PR DESCRIPTION
**Pull request by carlkyrillos**
# Description
Version bump to 1.7.0 that fixes the Endpoint Detailed dashboard that would incorrectly show the grafana UI endpoint as down. This bugfix adds a ServiceAccount with namespace:get permissions and adds the ServiceAccount's token to the Blackbox Exporter ConfigMap so the Blackbox probe requests can reach the grafana endpoint that is behind a proxy.

JIRA: https://issues.redhat.com/browse/MGDAPI-2282

## Verification
1. Create a new OSD cluster via [OpenShift](https://qaprodauth.cloud.redhat.com/openshift/) and target it once it has finished provisioning.
2. Pull [this PR](https://github.com/integr8ly/integreatly-operator/pull/2052) from the integreatly-operator repo and change [this line](https://github.com/integr8ly/integreatly-operator/blob/4fe1c29837037879fb806512cd9870ae4b6d2ac8/manifests/integreatly-monitoring/1.7.0/application-monitoring-operator.v1.7.0.clusterserviceversion.yaml#L238) to point to the image stored in my Quay [repository](https://quay.io/repository/ckyrillo/application-monitoring-operator):
`image: quay.io/ckyrillo/application-monitoring-operator:v1.7.0`
3. Install RHOAM from the command line to the remote cluster using the PR from Step 2.
4. Once the installation is complete, verify that the "grafana-ui UP/DOWN Status" panel in the Endpoint Dashboard from the redhat-rhoam-middleware-monitoring-operator grafana route correctly displays that the grafana service is UP.

Screenshot of the dashboard without the bugfix:
MGDAPI-2282-bug

Screenshot of the dashboard with the bugfix:
MGDAPI-2282-bugfix

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added a test case that will be used to verify my changes 
- [ ] Verified independently on a cluster by reviewer

### CODE REVIEW

File: deploy/cluster-roles/blackbox-exporter-clusterrole_binding.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+metadata:
+  name: blackbox-exporter
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: blackbox-exporter
+subjects:
+  - kind: ServiceAccount
+    name: blackbox-exporter-service-account
+    namespace: redhat-rhoam-middleware-monitoring-operator
```
Comment (matskiv) [inline deploy/cluster-roles/blackbox-exporter-clusterrole_binding.yaml:12]: Other CRB files reference `application-monitoring ` namespace, e.g.:
https://github.com/integr8ly/application-monitoring-operator/blob/c6fc97334e4e88eb3f05aef0cc4671a35e242a74/deploy/cluster-roles/alertmanager-clusterrole_binding.yaml#L11
OLM will then take care of updating this field during installation/upgrade.

File: pkg/controller/applicationmonitoring/applicationmonitoring_controller.go [hunk 2]
```diff
@@ -362,6 +363,36 @@ func (r *ReconcileApplicationMonitoring) reconcileBlackboxExporterConfig(cr *app
 	if r.extraParams == nil {
 		r.extraParams = map[string]string{}
 	}
+
+	// Add bearer_token to ConfigMap to allow Blackbox exporter requests to get through proxy in front of Grafana service
+	blackboxServiceAccount := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "blackbox-exporter-service-account",
+			Namespace: "redhat-rhoam-middleware-monitoring-operator",
```
Comment (matskiv) [inline pkg/controller/applicationmonitoring/applicationmonitoring_controller.go:371]: Namespace shouldn't be hard-coded. It should create it in a namespace where ApplicationMonitoring CR is created.

File: pkg/controller/applicationmonitoring/applicationmonitoring_controller.go [hunk 3]
```diff
@@ -362,6 +363,36 @@ func (r *ReconcileApplicationMonitoring) reconcileBlackboxExporterConfig(cr *app
...
+	var secretName string
+	for _, secret := range blackboxServiceAccount.Secrets {
+		if res, _ := regexp.MatchString("blackbox-exporter-service-account-token", secret.Name); res {
+			secretName = secret.Name
+		}
+	}
+	blackboxServiceAccountSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: secretName,
+			Namespace: "redhat-rhoam-middleware-monitoring-operator",
```
Comment (matskiv) [inline pkg/controller/applicationmonitoring/applicationmonitoring_controller.go:387]: Same as above, namespace shouldn't be hard-coded.
Here I would reference `blackboxServiceAccount.Namespace`

File: pkg/controller/applicationmonitoring/applicationmonitoring_controller.go (hunk 2 above)
Comment (matskiv) [reply pkg/controller/applicationmonitoring/applicationmonitoring_controller.go:371]: During creation you already do that with `cr.GetNamespace()` :)

File: deploy/cluster-roles/blackbox-exporter-clusterrole_binding.yaml (hunk 1 above)
Comment (carlkyrillos) [reply deploy/cluster-roles/blackbox-exporter-clusterrole_binding.yaml]: Fixed

File: pkg/controller/applicationmonitoring/applicationmonitoring_controller.go (hunk 2 above)
Comment (carlkyrillos) [reply pkg/controller/applicationmonitoring/applicationmonitoring_controller.go]: Fixed

File: pkg/controller/applicationmonitoring/applicationmonitoring_controller.go (hunk 3 above)
Comment (carlkyrillos) [reply pkg/controller/applicationmonitoring/applicationmonitoring_controller.go]: Fixed

**Comment by jackdelahunt** [comment]
- Installed rhoam with custom branch
 - grafana-ui UP/DOWN Status was all green

image

/lgtm

**Comment by carlkyrillos** [comment]
/assign @pb82
Could you take a look at this PR when you get a chance?

**Pull request merged**